### PR TITLE
(Fix) Updated the procedures and imaging apps to load the current day orders by default on page load

### PR DESCRIPTION
**Pull request by FelixKiprotich350**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
Updated the procedures and imaging apps to load the current day orders by default on page load. You can later adjsut the dates as you want. This helps in avoiding loading and unnecessary orders and improves efficiency

## Screenshots

Screenshot from 2024-12-02 12-21-25

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by ojwanganto** [comment]
Thanks, kongozi. We would like to have the same apply to all the tabs so that we consistently have efficiency across the tabs. I think the implementation in the lab module already has this. Please check if there is anything already that we can borrow.

File: packages/esm-imaging-orders-app/src/shared/ui/common/grouped-orders-table.component.tsx [hunk 1]
```diff
@@ -92,59 +90,70 @@ const GroupedOrdersTable: React.FC<GroupedOrdersTableProps> = (props) => {
...
-              />
+              <Layer className={styles.toolbarItem}>
+                <OrdersDateRangePicker />
+              </Layer>
+              <Layer className={styles.toolbarItem}>
+                <TableToolbarSearch
+                  expanded
+                  persistent={true}
+                  onChange={(event) => setSearchString(event.target.value)}
+                  placeholder={t('searchbypatientname', 'Search by patient name')}
```
Comment (makombe) [inline packages/esm-imaging-orders-app/src/shared/ui/common/grouped-orders-table.component.tsx:101]:
```suggestion
                  placeholder={t('searchByPatientName', 'Search by patient name')}
```

File: packages/esm-procedure-orders-app/src/shared/ui/common/grouped-orders-table.component.tsx [hunk 2]
```diff
@@ -92,59 +91,70 @@ const GroupedOrdersTable: React.FC<GroupedOrdersTableProps> = (props) => {
...
-              />
+              <Layer className={styles.toolbarItem}>
+                <OrdersDateRangePicker />
+              </Layer>
+              <Layer className={styles.toolbarItem}>
+                <TableToolbarSearch
+                  expanded
+                  persistent={true}
+                  onChange={(event) => setSearchString(event.target.value)}
+                  placeholder={t('searchbypatientname', 'Search by patient name')}
```
Comment (makombe) [inline packages/esm-procedure-orders-app/src/shared/ui/common/grouped-orders-table.component.tsx:102]:
```suggestion
                  placeholder={t('searchByPatientName', 'Search by patient name')}
```

File: packages/esm-procedure-orders-app/src/shared/ui/common/grouped-orders-table.scss [hunk 3]
```diff
@@ -11,3 +13,30 @@
...
+  align-items: center;
+
+  & p {
+    padding-right: 0.5rem;
+    @include type.type-style('body-01');
+    color: colors.$gray-70;
+  }
+
+  :global(.cds--date-picker__input) {
+    height: spacing.$spacing-09;
```
Comment (makombe) [inline packages/esm-procedure-orders-app/src/shared/ui/common/grouped-orders-table.scss:28]:
```suggestion
    height: layout.$spacing-09;
```

File: packages/esm-procedure-orders-app/src/shared/ui/common/grouped-orders-table.scss [hunk 4]
```diff
@@ -11,3 +13,30 @@
...
+  }
+
+  :global(.cds--dropdown) {
+    height: spacing.$spacing-09;
```
Comment (makombe) [inline packages/esm-procedure-orders-app/src/shared/ui/common/grouped-orders-table.scss:32]:
```suggestion
    height: layout.$spacing-09;
```

**Comment by FelixKiprotich350** [comment]
> same

Noted ill work on this.

File: packages/esm-imaging-orders-app/src/hooks/useOrdersWorklist.ts [hunk 5]
```diff
@@ -1,14 +1,22 @@
...
 import { useMemo, useCallback } from 'react';
 import { Result } from '../imaging-tabs/work-list/work-list.resource';
 import { ImagingConfig } from '../config-schema';
 import { FulfillerStatus } from '../shared/ui/common/grouped-imaging-types';
+import dayjs from 'dayjs';
+import { DateFilterContext } from '../types';
 
-const createApiUrl = (radiologyOrderTypeUuid: string, activatedOnOrAfterDate: string, fulfillerStatus: string) => {
+const createApiUrl = (
+  radiologyOrderTypeUuid: string,
```
Comment (donaldkibet) [inline packages/esm-imaging-orders-app/src/hooks/useOrdersWorklist.ts:11]:
```suggestion
  orderTypeUuid: string,
```

File: packages/esm-imaging-orders-app/src/hooks/useOrdersWorklist.ts [hunk 6]
```diff
@@ -17,10 +25,19 @@ export function useOrdersWorkList(activatedOnOrAfterDate: string, fulfillerStatu
...
-    () => createApiUrl(radiologyOrderTypeUuid, activatedOnOrAfterDate, fulfillerStatus),
-    [radiologyOrderTypeUuid, activatedOnOrAfterDate, fulfillerStatus],
+    () =>
+      createApiUrl(
+        radiologyOrderTypeUuid,
+        dateRange.at(0).toISOString(),
+        dateRange.at(1).toISOString(),
+        fulfillerStatus,
+      ),
+    [radiologyOrderTypeUuid, dateRange.at(0).toISOString(), dateRange.at(1).toISOString(), fulfillerStatus],
```
Comment (donaldkibet) [inline packages/esm-imaging-orders-app/src/hooks/useOrdersWorklist.ts:40]: Check this linting errors and fix them

**Review by donaldkibet — CHANGES_REQUESTED** [review]
Thanks @FelixKiprotich350, Great work. I final request could we have unit test to try prevent regression

**Review by donaldkibet — APPROVED** [review]

**Pull request merged**